### PR TITLE
Using DotNetCore version of BclRewriter in mscorlib build

### PR DESCRIPTION
**Pull request by joperezr**
Since we have updated the buildtools package to contain a .Net Core version of BclRewriter, now we can use that to build mscorlib

### CODE REVIEW

**Comment by joperezr** [comment]
cc @weshaggard @ellismg

File: src/NuGet.Config [hunk 1]
```diff
@@ -6,6 +6,9 @@
   <packageSources>
     <add key="myget.org dotnet-buildtools" value="https://www.myget.org/F/dotnet-buildtools/" />
     <add key="nuget.org" value="https://www.nuget.org/api/v2/" />
+    <add key="myget.org dotnet-core" value="https://www.myget.org/F/dotnet-core/" />
+    <add key="myget.org dotnet-coreclr" value="https://www.myget.org/F/dotnet-coreclr/" />
```
Comment (ellismg) [inline src/NuGet.Config]: Why did this feed get added (and dotnet-corefx?)  I thought these were dead.

File: src/mscorlib/Tools/BclRewriter/BclRewriter.targets [hunk 2]
```diff
@@ -5,26 +5,33 @@
...
   </PropertyGroup>
 
+  <Import Project="$(ToolsDir)toolruntime.targets" />
+  
   <ItemGroup>
     <RewrittenAssembly Include="$(BclRewriterOutput)" />
   </ItemGroup>
   
+  <PropertyGroup>
+    <BclRewriterCommand Condition="'$(BclRewriterCommand)'==''">"$(ToolRuntimePath)$(ToolHost)" "$(ToolsDir)BclRewriter.exe"</BclRewriterCommand>
```
Comment (ellismg) [inline src/mscorlib/Tools/BclRewriter/BclRewriter.targets]: I think build.sh needs to be passing something like ToolNugetRuntimeId as a property, the defaults are going to be wrong.

**Comment by ellismg** [comment]
Merging this is going to break the ability of @pgavlin to build mscorlib.dll on FC22, which is his primary development environment.

Could we update build tools to look and see if an environment variable, say BUILDTOOLS_TOOL_RUNTIME, is set.  If so, after deploying the runtime via nuget we copy all of the files from the directory it points to into the runtime?  Then folks can build their own native components for non ubuntu 14.04 environments.

File: src/NuGet.Config (hunk 1 above)
Comment (joperezr) [inline src/NuGet.Config]: right this is my bad, I believe the only one we really need is the dotnet-core one to be able to restore the required packages for tool-runtime

File: src/mscorlib/Tools/BclRewriter/BclRewriter.targets (hunk 2 above)
Comment (joperezr) [inline src/mscorlib/Tools/BclRewriter/BclRewriter.targets]: you are right I forgot to change that, although do you want me to do it on build.sh or should I do it in this file by checking if the os is ubuntu and overwriting the variable?

**Comment by ellismg** [comment]
@joperezr I don't think there's a good way to detect if you are on OSX vs Linux vs FreeBSD, etc in the MSBuild scripts.  I would prefer if we just let build.sh do the platform sniffing and rid generation and pass the RID to msbuild.

**Comment by jasonwilliams200OK** [comment]
@ellismg, another approach of platform sniffing can be using cmake set an environment variable like:

```cmake
SET(ENV{CORECLR_OS_NAME} CMAKE_SYSTEM_NAME)
```
Then use `$CORECLR_OS_NAME` subsequently.

**Comment by joperezr** [comment]
@ellismg To address the point you made about not breaking @pgavlin on FC22, I created PR dotnet/buildtools#301

File: src/mscorlib/Tools/BclRewriter/BclRewriter.targets [hunk 3]
```diff
@@ -5,30 +5,34 @@
   <!-- ********************************************************************************************* -->
 
   <PropertyGroup>
-    <BclRewriterCommand Condition="'$(BclRewriterCommand)'==''">$(CoreCLRToolsDir)\BclRewriter.exe</BclRewriterCommand>
+    <BaseOutputPath Condition="'$(BaseOutputPath)' == ''">$(__BinDir)</BaseOutputPath>
+    <!-- In Ubuntu 14.04 __BinDir is not set, so we need to use CMakeBinDir instead. -->
+    <BaseOutputPath Condition="'$(BaseOutputPath)' == ''">$(__CMakeBinDir)</BaseOutputPath>
```
Comment (akoeplinger) [inline src/mscorlib/Tools/BclRewriter/BclRewriter.targets]: might be easier to just pass `__BinDir` to msbuild in build.sh instead.

Comment (joperezr) [inline src/mscorlib/Tools/BclRewriter/BclRewriter.targets]: That is a good point, I'll fix that.

**Comment by joperezr** [comment]
Ping! :)

File: src/mscorlib/Tools/BclRewriter/BclRewriter.targets [hunk 4]
```diff
@@ -5,30 +5,32 @@
   <!-- ********************************************************************************************* -->
 
   <PropertyGroup>
-    <BclRewriterCommand Condition="'$(BclRewriterCommand)'==''">$(CoreCLRToolsDir)\BclRewriter.exe</BclRewriterCommand>
+    <IntermediateOutputRootPath Condition="'$(IntermediateOutputRootPath)' == ''">$(IntermediateOutputPath)</IntermediateOutputRootPath>
     <BclRewriterModelFile>$(MscorlibDir)model.xml</BclRewriterModelFile>
     <BclRewriterWorkDir>$(IntermediateOutputPath)\BclRewriter</BclRewriterWorkDir>
     <BclRewriterSymbolOutput>$(IntermediateOutputPath)\BclRewriter\$(TargetName).pdb</BclRewriterSymbolOutput>
     <BclRewriterOutput>$(IntermediateOutputPath)\BclRewriter\$(TargetName)$(TargetExt)</BclRewriterOutput>
+    <BuildToolsSemaphore>$(IntermediateOutputPath)\BclRewriter\semaphore</BuildToolsSemaphore>
```
Comment (weshaggard) [inline src/mscorlib/Tools/BclRewriter/BclRewriter.targets]: Looks like coreclr repo doesn't have this yet. Perhaps you can mimic it by using setting BuildToolsSemaphore to $(ToolsDir)Microsoft.DotNet.Build.Tasks.dll

File: src/mscorlib/Tools/BclRewriter/BclRewriter.targets [hunk 5]
```diff
@@ -5,30 +5,32 @@
   <!-- ********************************************************************************************* -->
 
   <PropertyGroup>
-    <BclRewriterCommand Condition="'$(BclRewriterCommand)'==''">$(CoreCLRToolsDir)\BclRewriter.exe</BclRewriterCommand>
+    <IntermediateOutputRootPath Condition="'$(IntermediateOutputRootPath)' == ''">$(IntermediateOutputPath)</IntermediateOutputRootPath>
```
Comment (weshaggard) [inline src/mscorlib/Tools/BclRewriter/BclRewriter.targets]: I think you should put IntermediateOutputRootPath and BuildToolsSempahore in the root dir.props as they aren't really bclrewriter specific.

**Comment by weshaggard** [comment]
LGTM other than the last couple comments

**Comment by joperezr** [comment]
@dotnet-bot test this please

**Pull request merged**